drivers/SmartThings/zigbee-smoke-detector/src/frient/init.lua

```diff
@@ -13,20 +13,254 @@
 -- limitations under the License.
 
 local battery_defaults = require "st.zigbee.defaults.battery_defaults"
+local capabilities = require "st.capabilities"
+local zcl_clusters = require "st.zigbee.zcl.clusters"
+local zcl_global_commands = require "st.zigbee.zcl.global_commands"
+local data_types = require "st.zigbee.data_types"
+local cluster_base = require "st.zigbee.cluster_base"
+local Basic = zcl_clusters.Basic
+local alarm = capabilities.alarm
+local smokeDetector = capabilities.smokeDetector
 
-local is_frient_smoke_detector = function(opts, driver, device)
-  if device:get_manufacturer() == "frient A/S" then
-    return true
+local IASWD = zcl_clusters.IASWD
+local IASZone = zcl_clusters.IASZone
+local TemperatureMeasurement = zcl_clusters.TemperatureMeasurement
+
+local ALARM_COMMAND = "alarmCommand"
+local ALARM_LAST_DURATION = "Custom_Alarm_Duration"
+local ALARM_DEFAULT_MAX_DURATION = 0x00F0
+local DEFAULT_WARNING_DURATION = 240
+local BATTERY_MIN_VOLTAGE = 2.3
+local BATTERY_MAX_VOLTAGE = 3.0
+
+local DEVELCO_MANUFACTURER_CODE = 0x1015
+local DEVELCO_BASIC_PRIMARY_SW_VERSION_ATTR = 0x8000
+
+local SIREN_ENDIAN = "siren_endian"
+local PRIMARY_SW_VERSION = "primary_sw_version"
+local SMOKE_ALARM_FIXED_ENDIAN_SW_VERSION = "040005"
+
+local TEMPERATURE_MEASUREMENT_ENDPOINT = 0x26
+
+local alarm_command = {
+  OFF = 0,
+  SIREN = 1
+}
+
+local CONFIGURATIONS = {
+  {
+    cluster = IASZone.ID,
+    attribute = IASZone.attributes.ZoneStatus.ID,
+    minimum_interval = 30,
+    maximum_interval = 300,
+    data_type = IASZone.attributes.ZoneStatus.base_type,
+    reportable_change = 1
+  },
+  {
+    cluster = TemperatureMeasurement.ID,
+    attribute = TemperatureMeasurement.attributes.MeasuredValue.ID,
+    minimum_interval = 60,
+    maximum_interval = 600,
+    data_type = TemperatureMeasurement.attributes.MeasuredValue.base_type,
+    reportable_change = 100
+  }
+}
+
+local function primary_sw_version_attr_handler(driver, device, value, zb_rx)
+  local primary_sw_version = value.value:gsub('.', function (c) return string.format('%02x', string.byte(c)) end)
+  device:set_field(PRIMARY_SW_VERSION, primary_sw_version, {persist = true})
+  if (primary_sw_version < SMOKE_ALARM_FIXED_ENDIAN_SW_VERSION) then
```

Comment:
The `<` operator for strings is alphabetical order, which is probably not what is intended here. I think you probably want to convert these to numbers first before working with them like this.